zcutil/make-release.py

```diff
@@ -123,17 +131,41 @@ def verify_releaseprev_tag(releaseprev):
         )
 
 
+@phase('Checking version.')
+def verify_version(release, releaseprev, hotfix):
+    if not hotfix:
+        return
+
+    expected = Version(
+        releaseprev.major,
+        releaseprev.minor,
+        releaseprev.patch,
+        releaseprev.betarc,
+        releaseprev.hotfix + 1 if releaseprev.hotfix else 1,
+    )
+    if release != expected:
+        raise SystemExit(
+            "Expected {!r}, given {!r}".format(
+                expected, release,
+            ),
+        )
+
+
 @phase('Initializing git.')
-def initialize_git(release):
+def initialize_git(release, hotfix):
     junk = sh_out('git', 'status', '--porcelain')
     if junk.strip():
         raise SystemExit('There are uncommitted changes:\n' + junk)
 
     branch = sh_out('git', 'rev-parse', '--abbrev-ref', 'HEAD').strip()
-    if branch != 'master':
+    if hotfix:
+        expected = 'hotfix-' + release.vtext
+    else:
+        expected = 'master'
+    if branch != expected:
         raise SystemExit(
-            "Expected branch 'master', found branch {!r}".format(
-                branch,
+            "Expected branch {!r}, found branch {!r}".format(
+                expected, branch,
             ),
         )
 
```

Comment:
Doing a test run of this script for a pretend hotfix. Am on a correctly named hotfix branch
On line 173, `git pull --ff-only` fails because I'm not on master now. 
Error:
```
There is no tracking information for the current branch.
Please specify which branch you want to merge with.
See git-pull(1) for details.

    git pull <remote> <branch>
```

Reply:
I think this will be fine. Per the hotfix release process, the correctly-named hotfix branch will be on the remote, and this call will succeed.

Reply:
Ok